graphjet-core/src/main/java/com/twitter/graphjet/algorithms/MultiThreadedPageRank.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright 2016 Twitter. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twitter.graphjet.algorithms;
+
+import com.google.common.util.concurrent.AtomicDoubleArray;
+import com.twitter.graphjet.bipartite.api.EdgeIterator;
+import com.twitter.graphjet.directed.api.OutIndexedDirectedGraph;
+import it.unimi.dsi.fastutil.longs.LongArrayList;
+import it.unimi.dsi.fastutil.longs.LongIterator;
+
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * <p>
+ * Multithreaded Implementation of PageRank. This implementation is the multi-threaded version of
+ * {@link PageRank}, inheriting its design and thus its limitations also.
```

Comment:
Added some comments. Let me know if you need any more details.

The short version is that we use atomic doubles so that it's okay that multiple thread are trying to updated the PageRank vector.